django_prices_openexchangerates/models.py

```diff
@@ -0,0 +1,79 @@
+from __future__ import unicode_literals
+
+from decimal import Decimal, ROUND_HALF_UP
+from threading import local
+
+from django.core.exceptions import ValidationError
+from django.db import models
+from django.utils.translation import ugettext_lazy as _
+from django.utils.encoding import python_2_unicode_compatible
+from django.conf import settings
+
+from .currencies import CURRENCIES
+
+
+CENTS = Decimal('0.01')
+
+_thread_locals = local()
+
+
+class CachingManager(models.Manager):
+
+    def get_rate(self, to_currency):
+        if not hasattr(_thread_locals, 'conversion_rates'):
+            _thread_locals.conversion_rates = {}
+        if to_currency not in _thread_locals.conversion_rates:
+            rates = self.all()
+            for rate in rates:
+                _thread_locals.conversion_rates[rate.to_currency] = rate
+        try:
+            rate = _thread_locals.conversion_rates[to_currency]
+        except KeyError:
+            rate = self.get(to_currency=to_currency)
+            _thread_locals.conversion_rates[to_currency] = rate
+        return rate
+
+
+@python_2_unicode_compatible
+class ConversionRate(models.Model):
```

Comment:
Please add a `__repr__` that includes the PK (and possibly currency and rate if that's not too much noise).